lib/Server.js

```diff
@@ -96,43 +85,41 @@ class Server {
 
     this.log = _log || createLogger(options);
 
-    // if the user enables http2, we can safely enable https
-    if (options.http2 && !options.https) {
-      options.https = true;
-    }
-
     this.originalStats =
-      options.stats && Object.keys(options.stats).length ? options.stats : {};
+      this.options.stats && Object.keys(this.options.stats).length
+        ? this.options.stats
+        : {};
 
-    this.hot = options.hot || options.hotOnly;
-    this.headers = options.headers;
-    this.progress = options.progress;
+    this.sockets = [];
+    this.contentBaseWatchers = [];
 
-    this.serveIndex = options.serveIndex;
+    // TODO this.<property> is deprecated (remove them in next major release.) in favor this.options.<property>
+    this.hot = this.options.hot || this.options.hotOnly;
+    this.headers = this.options.headers;
+    this.progress = this.options.progress;
 
-    this.clientOverlay = options.overlay;
-    this.clientLogLevel = options.clientLogLevel;
+    this.serveIndex = this.options.serveIndex;
 
-    this.publicHost = options.public;
-    this.allowedHosts = options.allowedHosts;
-    this.disableHostCheck = !!options.disableHostCheck;
+    this.clientOverlay = this.options.overlay;
+    this.clientLogLevel = this.options.clientLogLevel;
 
-    this.sockets = [];
+    this.publicHost = this.options.public;
+    this.allowedHosts = this.options.allowedHosts;
+    this.disableHostCheck = !!this.options.disableHostCheck;
 
-    if (!options.watchOptions) {
-      options.watchOptions = {};
+    if (!this.options.watchOptions) {
+      this.options.watchOptions = {};
     }
-    // ignoring node_modules folder by default
-    options.watchOptions.ignored = options.watchOptions.ignored || [
+    // Ignoring node_modules folder by default
+    this.options.watchOptions.ignored = this.options.watchOptions.ignored || [
       /node_modules/,
     ];
-    this.watchOptions = options.watchOptions;
+    this.watchOptions = this.options.watchOptions;
 
-    this.contentBaseWatchers = [];
     // Replace leading and trailing slashes to normalize path
     this.sockPath = `/${
-      options.sockPath
-        ? options.sockPath.replace(/^\/|\/$/g, '')
+      this.options.sockPath
+        ? this.options.sockPath.replace(/^\/|\/$/g, '')
```

Comment:
Now we should use `this.options.<option>` instead `this.<option>` because it is very uncomfortable and require extra code like: `this.clientOverlay = options.clientOverlay` so we should remove them in future major release